pkg/framework/plugins/removepodshavingtoomanyrestarts/toomanyrestarts.go

```diff
@@ -0,0 +1,139 @@
+	_ framework.DeschedulePlugin = &RemovePodsHavingTooManyRestarts{}
+)
+
+// New builds plugin from its arguments while passing a handle
+func New(args runtime.Object, handle framework.Handle) (framework.Plugin, error) {
+	tooManyRestartsArgs, ok := args.(*componentconfig.RemovePodsHavingTooManyRestartsArgs)
+	if !ok {
+		return nil, fmt.Errorf("want args to be of type RemovePodsHavingTooManyRestartsArgs, got %T", args)
+	}
+
+	var includedNamespaces, excludedNamespaces sets.String
+	if tooManyRestartsArgs.Namespaces != nil {
+		includedNamespaces = sets.NewString(tooManyRestartsArgs.Namespaces.Include...)
+		excludedNamespaces = sets.NewString(tooManyRestartsArgs.Namespaces.Exclude...)
+	}
+
+	podFilter, err := podutil.NewOptions().
+		WithFilter(handle.Evictor().Filter).
+		WithNamespaces(includedNamespaces).
+		WithoutNamespaces(excludedNamespaces).
+		WithLabelSelector(tooManyRestartsArgs.LabelSelector).
+		BuildFilterFunc()
+	if err != nil {
+		return nil, fmt.Errorf("error initializing pod filter function: %v", err)
+	}
+
+	podFilter = podutil.WrapFilterFuncs(podFilter, func(pod *v1.Pod) bool {
+		if err := validateCanEvict(pod, tooManyRestartsArgs); err != nil {
+			klog.V(4).InfoS(fmt.Sprintf("ignoring pod for eviction due to: %s", err.Error()), "pod", klog.KObj(pod))
+			return false
+		}
+		return true
+	})
+
+	return &RemovePodsHavingTooManyRestarts{
+		handle:    handle,
+		args:      tooManyRestartsArgs,
+		podFilter: podFilter,
+	}, nil
+}
+
+// Name retrieves the plugin name
+func (d *RemovePodsHavingTooManyRestarts) Name() string {
+	return PluginName
+}
+
+// Deschedule extension point implementation for the plugin
+func (d *RemovePodsHavingTooManyRestarts) Deschedule(ctx context.Context, nodes []*v1.Node) *framework.Status {
+	for _, node := range nodes {
+		klog.V(1).InfoS("Processing node", "node", klog.KObj(node))
+		pods, err := podutil.ListAllPodsOnANode(node.Name, d.handle.GetPodsAssignedToNodeFunc(), d.podFilter)
+		if err != nil {
+			// no pods evicted as error encountered retrieving evictable Pods
+			return &framework.Status{
+				Err: fmt.Errorf("error listing pods on a node: %v", err),
+			}
+		}
+		totalPods := len(pods)
+		for i := 0; i < totalPods; i++ {
+			d.handle.Evictor().Evict(ctx, pods[i], evictions.EvictOptions{})
+			if d.handle.Evictor().NodeLimitExceeded(node) {
+				break
+			}
+		}
+	}
+	return nil
+}
+
+// validateCanEvict looks at tooManyRestartsArgs to see if pod can be evicted given the args.
+func validateCanEvict(pod *v1.Pod, tooManyRestartsArgs *componentconfig.RemovePodsHavingTooManyRestartsArgs) error {
+	var errs []error
+
+	restarts := calcContainerRestartsFromStatuses(pod.Status.ContainerStatuses)
+	if tooManyRestartsArgs.IncludingInitContainers {
+		restarts += calcContainerRestartsFromStatuses(pod.Status.InitContainerStatuses)
+	}
+
+	if restarts < tooManyRestartsArgs.PodRestartThreshold {
+		errs = append(errs, fmt.Errorf("pod's restarts %v is excluded", restarts))
```

Comment:
There's at most a single error. No need to store the error into an array:
```
var err error
...
err = fmt.Errorf("pod's restarts %v is excluded", restarts)
...
return err
```
will do as well.

Reply:
s/"pod's restarts %v is excluded"/"number of container restarts (%v) not exceeding the threshold"